aep/general/0002/aep.md

```diff
@@ -0,0 +1,98 @@
+# AEP Numbering and Organization
+
+This document describes the numbering system and general organization of AEPs.
+
+AEP numbers:
+
+- Are unique. No two AEPs share the same number.
+- Indicate the order in which they are authored.
+
+To help with organization and discovery, AEPs also have a category and a slug
+to easily discovery and refer to the AEP.
+
+See below for an example of an AEP id, category, and slug, expressed via YAML
+front-matter:
+
+```yaml
+id: 11
+category: 'resources'
+slug: 'create'
+```
+
+## AEP Numbers
+
+### Core AEPs
+
+To uniquely identify an AEP, each one is assigned a permanent number as it is
+merged.
+
+- A merged AEP **must** use the next available number.
+- AEPs **must** be numbered between 1 and 9999.
+
+### Organization-specific AEPs
+
+- Organizations **may** extend the AEPs with guidance that is relevant to them.
+- Organization-specific AEPs **must** use AEP numbers 10000 or greater.
+
+## AEP Categories
+
+To help with discovery of an AEP, each one is assigned a category: a string
+that describes a theme or grouping the AEP belongs in.
+
+- The category of an AEP **may** change over time.
+
+As an example, the guidance around custom methods may grow to multiple AEPs. As
+such, AEPS may create a new category "custom-methods" and recategorize relevant
+AEPs (mutating custom methods, retrieval custom methods) under it.
+
+Categories are expressed in the YAML front-matter of an AEP:
+
+```yaml
+category: 'custom-methods'
+```
+
+## AEP Slugs
+
+Each AEP is given a slug: a string by which it can be easily referenced.
```

Comment:
Are slugs also used for short links, e.g. aep.dev/create?

Reply:
I was thinking of adding that once I do work in the site-generator to support it, but I don't mind adding that now. WDYT?

If we did, I think `aep.dev/{slug}` makes sense. Although we may need to make sure slug names can't be used (e.g. `category`).

I'd kind of wonder if we want to namespace this, with something like `aep.dev/p/create` or something like that. A little less aesthetically pleasing, but works around a giant headache?

Reply:
Might as well write it now IMO, it's not like the site generator is in tightly sync with this repo anyway.  For now we can implement it as a matter of policy - the shortlink (already supported by the site generator) must match the slug.

I think the `/p/` gets too much in the way of typing out shortlinks like `aip.dev/create`.  Better to namespace categories, if we're worried about collisions, IMO - they'll probably be much less frequently referenced by manually typed links.

Reply:
ack, will add it before I merge.

I was more concerned about having the set of possible slugs be everything, except for a set of constants (e.g. "categories" so we don't conflict with them). 

Just playing around with this a bit, I think we're looking at:

```
aep.dev/ # landing page
aep.dev/create # slug to an AEP
aep.dev/2 # the actual AEP.
aep.dev/categories/ # category page
aep.dev/categories/resources # page listing AEPs for a category
```

Reply:
SGTM, approving to unblock the PR whenever you make that change.

Reply:
added!